internal/core/metadata/v2/controller/http/deviceprofile.go

```diff
@@ -0,0 +1,145 @@
+package http
+
+import (
+	"net/http"
+
+	"github.com/edgexfoundry/edgex-go/internal/core/metadata/v2/application"
+	"github.com/edgexfoundry/edgex-go/internal/core/metadata/v2/io"
+	"github.com/edgexfoundry/edgex-go/internal/pkg"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/correlation"
+
+	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/di"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos"
+	commonDTO "github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/common"
+	requestDTO "github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/requests"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+)
+
+const (
+	// URL PATH
+	UPLOADFILE = "/uploadfile"
+)
+
+type DeviceProfileController struct {
+	reader io.DeviceProfileReader
+	dic    *di.Container
+}
+
+// NewDeviceProfileController creates and initializes an DeviceProfileController
+func NewDeviceProfileController(dic *di.Container) *DeviceProfileController {
+	return &DeviceProfileController{
+		reader: io.NewDeviceProfileRequestReader(),
+		dic:    dic,
+	}
+}
+
+func (dc *DeviceProfileController) AddDeviceProfile(w http.ResponseWriter, r *http.Request) {
+	if r.Body != nil {
+		defer func() { _ = r.Body.Close() }()
+	}
+
+	lc := container.LoggingClientFrom(dc.dic.Get)
+
+	ctx := r.Context()
+
+	addDeviceProfileDTOs, err := dc.reader.ReadAddDeviceProfileRequest(r.Body, &ctx)
+	if err != nil {
+		lc.Error(err.Error())
+		lc.Debug(err.DebugMessages())
+		http.Error(w, err.Message(), err.Code())
+		return
+	}
+	deviceProfiles := requestDTO.AddDeviceProfileReqToDeviceProfileModels(addDeviceProfileDTOs)
+
+	// map DeviceProfile models to AddDeviceProfileResponse DTOs
+	var addResponses []interface{}
+	for i, d := range deviceProfiles {
+		newId, err := application.AddDeviceProfile(d, ctx, dc.dic)
+		var addDeviceProfileResponse interface{}
+		// get the requestID from AddDeviceProfileDTO
+		reqId := addDeviceProfileDTOs[i].RequestID
+
+		if err == nil {
+			addDeviceProfileResponse = commonDTO.NewBaseWithIdResponse(
+				reqId,
+				"Add device profiles successfully",
+				http.StatusCreated,
+				newId)
+		} else {
+			lc.Error(err.Error())
+			lc.Debug(err.DebugMessages())
+			addDeviceProfileResponse = commonDTO.NewBaseResponse(
+				reqId,
+				err.Error(),
+				err.Code())
+		}
+		addResponses = append(addResponses, addDeviceProfileResponse)
+	}
+
+	w.Header().Set(clients.ContentType, clients.ContentTypeJSON)
+	w.Header().Set(clients.CorrelationHeader, correlation.FromContext(ctx))
+	w.WriteHeader(http.StatusMultiStatus)
+	pkg.Encode(addResponses, w, lc)
```

Comment:
Added comment.

Reply:
@lenny-intel 
We use the pkg.Encode() in many function like:
https://github.com/edgexfoundry/edgex-go/blob/master/internal/core/data/router.go#L614

Do we really need to add the comment or refactor the function name?

Reply:
:-) I would prefer to refactor name as you indicate it is used many places and feels silly to keep add comment. @cloudxxx8 , you ok with this since is is more than V2 scope? It is just a name refactor... How about `EncodeAndSend`?